packages/shared/src/hooks/profile/useProfile.ts

```diff
@@ -0,0 +1,25 @@
+import { useContext } from 'react';
+import { useQuery } from '@tanstack/react-query';
+import { getProfile, PublicProfile } from '../../lib/user';
+import AuthContext from '../../contexts/AuthContext';
+import { generateQueryKey, RequestKey } from '../../lib/query';
+
+export function useProfile(initialUser?: PublicProfile): PublicProfile {
+  const { user: loggedUser } = useContext(AuthContext);
+  const isSameUser = loggedUser?.id === initialUser?.id;
+
+  const { data: user } = useQuery(
+    generateQueryKey(RequestKey.Profile, initialUser),
+    () => getProfile(initialUser?.id),
+    {
+      placeholderData: initialUser,
```

Comment:
This initialUser is already fetched data right, I think you might want to use `initialData` in that case?
It's a small nuance, but initial will persist already.

https://tkdodo.eu/blog/placeholder-and-initial-data-in-react-query

Reply:
My default is initial data but here I specifically wanted placeholder to make sure I fetch fresh data for a specific use case. Otherwise, I may have shown cached data to the user